sklearn/preprocessing/_data.py

```diff
@@ -771,6 +790,11 @@ def partial_fit(self, X, y=None):
             counts_nan = sparse_constructor(
                         (np.isnan(X.data), X.indices, X.indptr),
                         shape=X.shape).sum(axis=0).A.ravel()
+            if sample_weight is not None:
+                # _sample_weight_mean = np.mean(sample_weight)
+                sample_weight = (sparse.csr_matrix(sample_weight)
+                                 if X.format == 'csr'
+                                 else sparse.csc_matrix(sample_weight))
```

Comment:
sample_weight should stay a numpy array. No need to make it sparse even if X is sparse